scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -1169,37 +1169,37 @@ private void build() {
 
     private Boolean isExecutionCalendarGIDefined(String jobDescriptor) {
         Document dom = XMLParser.parse(jobDescriptor);
-        dom.getDocumentElement().normalize();
-
         Boolean exists = false;
         Boolean executionCalendarDefined = false;
-
         NodeList genericInfo = dom.getElementsByTagName("genericInformation");
-        // get the first item
-        Node root = genericInfo.item(0);
-        NodeList list = root.getChildNodes();
-        for (int i = 0; i < list.getLength(); i++) {
-            Node node = list.item(i);
-            if (node.getNodeName().equals("info") && node.hasAttributes()) {
-                NamedNodeMap attributes = node.getAttributes();
-                for (int j = 0; j < attributes.getLength(); j++) {
-                    Node attribute = attributes.item(j);
-                    if (attribute.getNodeType() == Node.ATTRIBUTE_NODE && attribute.getNodeName().equals("name") &&
-                        attribute.getNodeValue().equalsIgnoreCase("execution_calendars")) {
-                        exists = true;
-                    }
-                    if (isAttributeExecCalendarValueDefined(attribute, "value") && exists) {
-                        if (!attribute.getNodeValue().isEmpty() && exists) {
-                            executionCalendarDefined = true;
-                            if (!attribute.getNodeValue().isEmpty()) {
-                                isExecCalendarValueNull = false;
-                            } else {
-                                isExecCalendarValueNull = true;
+        // check if the job has genericInformation or not
+        if (genericInfo != null && genericInfo.getLength() > 0) {
```

Comment:
is it possible to refactor this big method into something more readable? maybe even separate this logic in a separate class..

Reply:
@marcocast I will address your comment in the next commit. Thanks a lot!